rust/src/datatypes/field.rs

```diff
@@ -330,11 +330,46 @@ impl Field {
 
     /// Merge the children of other field into this one.
     pub(super) fn merge(&mut self, other: &Self) -> Result<()> {
-        for other_child in other.children.as_slice() {
-            if let Some(field) = self.child_mut(&other_child.name) {
-                field.merge(other_child)?;
-            } else {
-                self.children.push(other_child.clone());
+        match (self.data_type(), other.data_type()) {
+            (DataType::Struct(_), DataType::Struct(_)) => {
+                for other_child in other.children.as_slice() {
+                    if let Some(field) = self.child_mut(&other_child.name) {
+                        field.merge(other_child)?;
+                    } else {
+                        self.children.push(other_child.clone());
+                    }
+                }
+            }
+            (DataType::List(_), DataType::List(_))
+            | (DataType::LargeList(_), DataType::LargeList(_)) => {
+                self.children[0].merge(&other.children[0])?;
+            }
+            (
+                DataType::FixedSizeList(_, self_list_size),
```

Comment:
can do 

```
DataType::FixedSizeList(_, self_size), DataType::FixedSizeList(_, other_size) if self_size == other_size {
    // Do nothing
}
```

So that the error message will just be handled by the default branch?